### PR TITLE
Add RevertToPageRevisionAction class and Admin API for reverting to a previous revision

### PR DESCRIPTION
**Pull request by Tijani-Dia**
See #7682.

* Do the tests still pass? Yes
* Does the code comply with the style guide? Yes
* For Python changes: Have you added tests to cover the new/fixed behaviour? Not yet

### CODE REVIEW

File: wagtail/admin/api/actions/revert_to_page_revision.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from wagtail.core.actions.revert_to_page_revision import (
+    RevertToPageRevisionAction,
+    RevertToPageRevisionError,
+)
+
+from .base import APIAction
+
+
+class RevertToPageRevisionAPIActionSerializer(Serializer):
+    previous_revision_id = fields.IntegerField()
```
Comment (kaedroho) [inline wagtail/admin/api/actions/revert_to_page_revision.py:18]: Maybe name this to ``revision_id`` (I think the ``previous_`` bit is implied from the action)

File: wagtail/core/actions/revert_to_page_revision.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    """
+
+    pass
+
+
+class RevertToPageRevisionAction:
+    def __init__(
+        self,
+        page,
+        previous_revision,
```
Comment (kaedroho) [inline wagtail/core/actions/revert_to_page_revision.py:24]: Similar to the API, we could rename this to ``revision``

**Comment by zerolab** [comment]
@Tijani-Dia is this still a draft?
note that it needs rebasing on main regardless of status

**Review by kaedroho — COMMENTED** [review]
I'm happy with this approach so far, just a suggestion on a name change. Thanks!

**Comment by Tijani-Dia** [comment]
@zerolab I was unsure of the approach. I will add some tests and mark it as ready for review.

I've rebased it to main, thanks!

File: wagtail/core/actions/revert_to_page_revision.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        self.submitted_for_moderation = submitted_for_moderation
+        self.approved_go_live_at = approved_go_live_at
+        self.changed = changed
+        self.clean = clean
+
+    def check(self, skip_permission_checks=False):
+        if self.page.alias_of_id:
+            raise RevertToPageRevisionError(
+                "Revisions are not required for alias pages as they are an exact copy of another page."
+            )
```
Comment (kaedroho) [inline wagtail/core/actions/revert_to_page_revision.py:45]: Could we add another check in here to make sure that the given revision is for the given page?

**Review by kaedroho — CHANGES_REQUESTED** [review]
Thanks again! I've just spotted that there's nowhere we are currently checking that the given revision is for the page (which probably didn't matter much in the past, but I think we should check now that this is exposed as an API).

File: wagtail/core/actions/revert_to_page_revision.py (hunk 3 above)
Comment (Tijani-Dia) [reply wagtail/core/actions/revert_to_page_revision.py:45]: Users submit the revision id and the revision is taken from the following (`instance.revisions`):
```  
def _action_from_data(self, instance, data):
        revision = get_object_or_404(instance.revisions, id=data["revision_id"])

        return RevertToPageRevisionAction(
            page=instance, revision=revision, user=self.request.user
        )
 ```
 
 So if we have a revision when the action object is instantiated from the API, we can be sure that we have a valid revision for the given page...?

Comment (kaedroho) [reply wagtail/core/actions/revert_to_page_revision.py:45]: Great, thanks! This looks good to me, sorry I missed that!

**Review by kaedroho — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**